target/linux/ipq40xx/base-files/etc/board.d/02_network

```diff
@@ -117,6 +117,9 @@ ipq40xx_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0u@eth0" "1:lan" "2:lan" "3:lan" "4:lan" "0u@eth1" "5:wan"
 		;;
+	tp-link,deco-m5-eu-v3.0)
```

Comment:
Vendor prefix for TP-Link is tplink in the compatible. You can make your and our job much easier by having a look at what other people have done in similar cases before you ...

Reply:
And for the particular setup here, this can be merged with the proper case above.